raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -90,223 +101,207 @@ def test_connect_peers():
-    payload = "marco"
-    client_1.send_message(payload, address_2)
+    payload = "echo"
+    client.send_message(payload, client.rsk_address.address)
     for resp in sub:
         received_message = notification_to_payload(resp)
         assert received_message == payload
         break  # only 1 message is expected
 
-    # now subscribe from node 1 to 2
-    client_1.subscribe_to(address_2)
 
-    # message should be successfully sent and received again
-    payload = "polo"
-    client_1.send_message(payload, address_2)
-    for resp in sub:
-        received_message = notification_to_payload(resp)
-        assert received_message == payload
-        break  # only 1 message is expected
+@pytest.mark.parametrize("nodes_to_clients", [{1: 1, 2: 1}, {1: 2}])
+def test_send_message_peers(comms_clients):
+    client_1 = comms_clients[1]
+    client_2 = comms_clients[2]
 
+    # subscribe both nodes to self to start listening
+    _, sub_1_ = client_1.subscribe_to(client_1.rsk_address.address)
+    _, sub_2 = client_2.subscribe_to(client_2.rsk_address.address)
 
-def test_send_message_shutdown():
-    # the comms_nodes fixture is not used in order to shut down nodes manually
-    nodes = []
-    try:
-        node_1 = CommsNode(CommsConfig(node_number=1, auto_connect=False))
-        nodes.append(node_1)
-        client_1, address_1 = node_1.client, node_1.address
-
-        node_2 = CommsNode(CommsConfig(node_number=2, auto_connect=False))
-        nodes.append(node_2)
-        client_2, address_2 = node_2.client, node_2.address
+    payload_1 = "hello from 1"
+    payload_2 = "hello from 2"
 
-        # connect clients
-        client_1.connect()
-        client_2.connect()
+    # send messages
+    client_1.send_message(payload_1, client_2.rsk_address.address)
+    client_2.send_message(payload_2, client_1.rsk_address.address)
 
-        # shut down node 2
-        node_2.stop()
+    for resp in sub_1_:
+        received_message = notification_to_payload(resp)
+        assert received_message == payload_2
+        break  # only 1 message is expected
 
-        # send message
-        client_1.send_message("into the void", address_2)  # no exception should be raised
-    finally:
-        for node in nodes:
-            # because node 2 can be already stopped, we cannot call stop() indiscriminately
-            if not node._process.poll():  # if True, node is still running
-                node.stop()
+    for resp in sub_2:
+        received_message = notification_to_payload(resp)
+        assert received_message == payload_1
+        break  # only 1 message is expected
 
 
-@pytest.mark.parametrize("amount_of_nodes", [1])
+@pytest.mark.parametrize("nodes_to_clients", [{1: 1}])
 @pytest.mark.xfail(reason="exceptions are not raised from comms node")
-def test_unsubscribe_from_invalid(comms_nodes):
-    client, address = comms_nodes[1].client, comms_nodes[1].address
+def test_unsubscribe_from_invalid(comms_clients):
```

Comment:
i think we should be consistent with the strategy adopted in `test_subscribe_to_invalid` here

Reply:
can you give me more info?

Reply:
> can you give me more info?

sorry, of course.

we are testing subscriptions to invalid addresses on a cluster with 2 nodes. we should analogously test for unsubscriptions from invalid addresses on the same topology too.

Reply:
agree, added